### PR TITLE
Fix generate type dialog invoked on type argument in a base list

### LINKED ISSUE #63280: Generate new type dialog from struct interface list as generic type parameter only allows generating interfaces
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Generate-new-type-dialog-from-struct-int/10113798)._

---
Consider the following C# code:

``` cs
using System.Collections.Generic;
namespace Test
{
    public readonly struct ExampleCollection : IEnumerable
    {
    }
}
```
Provided that the type `Example` does not exist, if one now moves the cursor to `Example` and invokes Quick Actions, there are three options for fixing it:

1. Generate class 'Example' in new file
1. Generate class 'Example'
1. Generate new type...

The first two options would create a code snippet of `public class Example`. Selecting the third option should allow generating other kinds of types. However, in this particular case, the only kind available within that dialog box is `interface`. It does not allow creating a `class`, `struct`, or `enum` contrary to user expectation.

**Expected behavior**: I can choose to create `Example` as a `class`, `struct`, `enum`, or `interface`.

**Actual behavior**: I can only choose to create `Example` as an `interface`.

---
#### Feedback Bot on 8/8/2022, 07:29 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #63280

### CODE REVIEW

File: src/EditorFeatures/DiagnosticsTestUtilities/Diagnostics/AbstractUserDiagnosticTest_GenerateTypeDialog.cs [hunk 1]
```diff
@@ -170,9 +170,9 @@ public abstract partial class AbstractUserDiagnosticTest
 
                 if (assertGenerateTypeDialogOptions != null)
                 {
-                    Assert.True(assertGenerateTypeDialogOptions.IsPublicOnlyAccessibility == generateTypeDialogOptions.IsPublicOnlyAccessibility);
-                    Assert.True(assertGenerateTypeDialogOptions.TypeKindOptions == generateTypeDialogOptions.TypeKindOptions);
-                    Assert.True(assertGenerateTypeDialogOptions.IsAttribute == generateTypeDialogOptions.IsAttribute);
+                    Assert.Equal(assertGenerateTypeDialogOptions.IsPublicOnlyAccessibility, generateTypeDialogOptions.IsPublicOnlyAccessibility);
+                    Assert.Equal(assertGenerateTypeDialogOptions.TypeKindOptions, generateTypeDialogOptions.TypeKindOptions);
+                    Assert.Equal(assertGenerateTypeDialogOptions.IsAttribute, generateTypeDialogOptions.IsAttribute);
```
Comment (Youssef1313) [inline src/EditorFeatures/DiagnosticsTestUtilities/Diagnostics/AbstractUserDiagnosticTest_GenerateTypeDialog.cs:175]: This produces better assertion failures.

**Comment by Youssef1313** [comment]
Can someone review this PR please? Thanks!

**Review by jmarolf — APPROVED** [review]

File: src/EditorFeatures/CSharpTest/Diagnostics/GenerateType/GenerateTypeTests_Dialog.cs [hunk 2]
```diff
@@ -2309,6 +2309,105 @@ public struct Bar
 isNewFile: false,
 assertGenerateTypeDialogOptions: new GenerateTypeDialogOptions(false, TypeKindOptions.Class | TypeKindOptions.Structure, false));
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateType)]
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Diagnostics/GenerateType/GenerateTypeTests_Dialog.cs:2313]: don't need the trait.  should already be on the type.

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Comment by CyrusNajmabadi** [comment]
@Youssef1313 do you want to bring this up to main so it can be merged?

**Comment by Youssef1313** [comment]
Sure. Let's see if it will turn green.

**Comment by Youssef1313** [comment]
@CyrusNajmabadi This is green.

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Pull request merged**